ambry-replication/src/main/java/com.github.ambry.replication/ReplicationMetrics.java

```diff
@@ -228,138 +230,142 @@ public ReplicationMetrics(MetricRegistry registry, List<? extends ReplicaId> rep
         registry.counter(MetricRegistry.name(ReplicaThread.class, "IntraColoReplicaThreadThrottleCount"));
     interColoReplicaThreadThrottleCount =
         registry.counter(MetricRegistry.name(ReplicaThread.class, "InterColoReplicaThreadThrottleCount"));
+    remoteReplicaInfoRemoveError =
+        registry.counter(MetricRegistry.name(ReplicaThread.class, "RemoteReplicaInfoRemoveError"));
+    remoteReplicaInfoAddError = registry.counter(MetricRegistry.name(ReplicaThread.class, "RemoteReplicaInfoAddError"));
     this.registry = registry;
     this.replicaLagInBytes = new ArrayList<Gauge<Long>>();
     populateInvalidMessageMetricForReplicas(replicaIds);
   }
 
   /**
-   * Updates per colo metrics for each thread pool
-   * @param datacenters List of datacenters to replicate from
+   * Updates given colo metrics.
+   * @param datacenter The datacenter to replicate from.
    */
-  public void populatePerColoMetrics(Set<String> datacenters) {
+  public void populateSingleColoMetrics(String datacenter) {
+    Meter interColoReplicationBytesRatePerDC =
+        registry.meter(MetricRegistry.name(ReplicaThread.class, "Inter-" + datacenter + "-ReplicationBytesRate"));
```

Comment:
we can do this in future prs, but I think it would simplify things a lot if we created a `DatacenterMetrics` class to group all of these metrics together. This would allow avoid the need to create separate maps for every single metric in favor of a map from datacenter name to `DatacenterMetrics` object.

Reply:
Will do it in follow up PRs. Other comments are addressed.